content/docs/hooks-faq.md

```diff
@@ -18,126 +18,126 @@ This page answers some of the frequently asked questions about [Hooks](/docs/hoo
 
-### What do Hooks mean for popular APIs like Redux `connect()` and React Router? {#what-do-hooks-mean-for-popular-apis-like-redux-connect-and-react-router}
+### Co hooki oznaczają dla popularnych API takich jak Redux connect() i React Router?{#what-do-hooks-mean-for-popular-apis-like-redux-connect-and-react-router}
 
-You can continue to use the exact same APIs as you always have; they'll continue to work.
+Możesz używać tych samych API, co do tej pory; będą nadal działać.
 
-In the future, new versions of these libraries might also export custom Hooks such as `useRedux()` or `useRouter()` that let you use the same features without needing wrapper components.
+W przyszłości nowe wersje tych bibliotek mogą również eksportować spersonalizowane hooki, takie jak `useRedux()` czy `useRouter()`, które pozwolą na użycie tych samych funkcjonalności bez potrzeby opakowywania komponentów.
 
-### Do Hooks work with static typing? {#do-hooks-work-with-static-typing}
+### Czy hooki współpracują ze statycznym typowaniem? {#do-hooks-work-with-static-typing}
 
-Hooks were designed with static typing in mind. Because they're functions, they are easier to type correctly than patterns like higher-order components. The latest Flow and TypeScript React definitions include support for React Hooks.
+Hooki zostały zaprojektowane z myślą o statycznym typowaniu. Dzięki temu że są funkcjami, łatwiej jest je poprawnie otypować w odróżnieniu od wzorców jak na przykład komponenty wyższego rzędu. Najnowsze definicje Reacta dla Flow i TypeScripta wspierają hooki.
 
-Importantly, custom Hooks give you the power to constrain React API if you'd like to type them more strictly in some way. React gives you the primitives, but you can combine them in different ways than what we provide out of the box.
+Co ważne, spersonalizowane hooki dają możliwość ograniczenia API Reacta, jeżeli tylko zechcesz możesz je otypować bardziej ściśle. React udostępnia podstawy, ale możesz je łączyć na różne sposoby, odmienne od tych które dostarczyliśmy w standardzie.
 
-### How to test components that use Hooks? {#how-to-test-components-that-use-hooks}
+### Jak testować komponenty które używają hooków? {#how-to-test-components-that-use-hooks}
 
-From React's point of view, a component using Hooks is just a regular component. If your testing solution doesn't rely on React internals, testing components with Hooks shouldn't be different from how you normally test components.
+Z punktu widzenia Reacta, komponent który wykorzystuje hooki jest zwyczajnym komponentem. Jeżeli rozwiązanie które testujesz, nie opiera się na wnętrzu Reacta, to podejście dotyczące testowania komponentów, które używają hooków, nie powinno się różnić od tego jak robisz to zazwyczaj.
```

Comment:
```suggestion
Z punktu widzenia Reacta, komponent wykorzystujący hooki jest zwyczajnym komponentem. Jeżeli twoje rozwiązanie do testów nie opiera się na wewnętrznej implementacji Reacta, to testowanie komponentów, które używają hooków, nie powinno różnić się od tego co robisz to zazwyczaj.
```
Hmm, _rozwiązanie które testujesz_ byłoby _solution that you test_. _Your testing solution_ to _Rozwiązanie do testowania_. Ale przyznam szczerze, że nie rozumiem co autor miał na myśli.